src/DataManagement/Saving/AlternativeSavingCalculator.cs

```diff
@@ -30,9 +29,23 @@ public CalculationResults GetSuggestedExpensesOffers(IEntryList entryList, IGoal
             var averagesOfAmountByImportanceDefault = new decimal[enumCount];
             var entryAmounts = new int[enumCount];
 
+            if (entryList.Count == 0)
+            {
+                generateRandomData = true;              
+            }
+
             for (var i = enumCount; i > (int)Importance.Necessary; i--)
             {
-                var expenses = entryList.GetBy((Importance)i);
+                IEntryList expenses;
+                if (!generateRandomData)
+                {
+                    expenses = entryList.GetBy((Importance)i);
+                }
+                else
+                {
+                    expenses = expensesRandomList.GetBy((Importance)i);
+                }
```

Comment:
This if else can be changed to '?' statement:

         expenses = !generateRandomData ? entryList.GetBy((Importance)i) : expensesRandomList.GetBy((Importance)i);